experimenter/manifesttool/appconfig.py

```diff
@@ -19,24 +18,13 @@ class Repository(BaseModel):
     name: str | None = None
     default_branch: str = Field(default="main")
 
-    @model_validator(mode="before")
-    @classmethod
-    def validate_default_branch(cls, values: dict[str, Any]):
-        ty = values.get("type")
-        default_branch = values.get("default_branch")
-
-        if ty == RepositoryType.HGMO and default_branch is None:
-            raise ValueError("hg.mozilla.org-hosted repositories require default_branch")
-
-        return values
-
     @model_validator(mode="before")
     @classmethod
     def validate_name(cls, values: dict[str, Any]):
-        _type = values.get("type")
+        ty = values.get("type")
         name = values.get("name")
 
-        if _type != RepositoryType.LOCAL and name is None:
+        if ty != RepositoryType.LOCAL and name is None:
```

Comment:
So annoying that `type` is a reserved word, just saw a pr yesterday that had the same issue.  Another pattern is to prefix it with some word like `repo_type` `repo_name` or something.